[NC | NSFS | Glacier] handle `ENOENT` in `_finalize_restore` and fix glacier tests not running

tangledbytes opened the pull request:
### Explain the changes
This PR simply adds the handler in `_finalize_restore` to make sure that it doesn't errors out if encounters `ENOENT`.

Fixes #8251

### Testing Instructions:
1. `./node_modules/.bin/mocha src/test/unit_tests/test_nsfs_glacier_backend.js`


- [ ] Doc added/updated
- [ ] Tests added

tangledbytes committed:
handle ENOENT in _finalize_restore

Signed-off-by: Utkarsh Srivastava <srivastavautkarsh8097@gmail.com>

handle ENOENT in open

Signed-off-by: Utkarsh Srivastava <srivastavautkarsh8097@gmail.com>

guymguym commented on src/sdk/nsfs_glacier_backend/tapecloud.js:
```diff
@@ -396,6 +396,13 @@ class TapeCloudGlacierBackend extends GlacierBackend {
 
             await fh.replacexattr(fs_context, undefined, GlacierBackend.XATTR_RESTORE_REQUEST);
         } catch (error) {
+            // restore does check if a file exist or not before triggering eeadm call but the file could get deleted
+            // after `restore` performs that check so check it here once again
+            if (error.code === 'ENOENT') {
+                dbg.warn(`TapeCloudGlacierBackend._finalize_restore: log entry unexpectedly not found: ${entry.path}`);
+                return;
+            }
```
it would be better if this was in a separate try-catch which is only on the open() call, to make sure we don't add any other fs calls in the flow that would be interpreted as missing file accidentally

tangledbytes replied:
Updated the PR

tangledbytes committed:
fix nsfs_glacier_backend tests not running

Signed-off-by: Utkarsh Srivastava <srivastavautkarsh8097@gmail.com>

tangledbytes commented:
@guymguym, I have added 2 commits in this PR. One PR is the actual fix while the other one fixes the glacier tests not running (my guess since about 3 weeks now). I have split the commits for the ease of backport at least as one would be picked up by DBS3.
